src/main/java/seedu/duke/userinterface/InputParser.java

```diff
@@ -1,4 +1,71 @@
 package seedu.duke.userinterface;
 
+import seedu.duke.exceptions.IncorrectDeadlineFormatException;
+import seedu.duke.exceptions.TaskTitleException;
+import seedu.duke.exceptions.TaskWrongFormatException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import static seedu.duke.userinterface.command.Add.DEADLINE_DELIMITER;
+import static seedu.duke.userinterface.command.Add.TASK_DELIMITER;
+
 public class InputParser {
+    public static String parseTaskTitle(String input) throws TaskWrongFormatException, TaskTitleException {
+        if (input.startsWith(TASK_DELIMITER)) {
+            String taskTitle = input.replace(TASK_DELIMITER, "");
+            if (taskTitle.isBlank()) {
+                throw new TaskTitleException();
+            }
+            int indexPos = taskTitle.indexOf("/by");
+            taskTitle = taskTitle.substring(0, indexPos).trim();
+            return taskTitle;
+        } else {
+            throw new TaskWrongFormatException();
+        }
+    }
+
+    /**
+     * Parses user's input to extract deadline.
+     *
+     * @param input input from user which contains the deadline.
+     * @return deadline
+     * @throws IncorrectDeadlineFormatException when the deadline input is in the wrong format.
+     * @throws TaskWrongFormatException when the deadline input is blank.
+     */
+    public static String parseDeadline(String input) throws Exception {
+        int dividerPos = input.indexOf(DEADLINE_DELIMITER);
+        input = input.substring(dividerPos);
+        if (input.startsWith(DEADLINE_DELIMITER)) {
+            String deadline = input.replace(DEADLINE_DELIMITER, "");
+            if (deadline.isBlank()) {
+                throw new TaskWrongFormatException();
+            }
+            if (!correctTimeFormat(deadline)) {
+                throw new IncorrectDeadlineFormatException();
+            }
+            return deadline;
+        } else {
+            throw new IncorrectDeadlineFormatException();
+        }
+    }
+
+    /**
+     * Checks if [deadline] input by the user is in the correct format.
+     *
+     * @param by is the string containing the deadline's due date and time.
+     * @return true when the input is in the correct format, otherwise false.
+     * @throws DateTimeParseException thrown when the user's input is in the wrong format.
```

Comment:
The function doesn't actually throw any exceptions though, as confirmed by the method signature, right?

Reply:
...Right. I'll delete that line.